src/test/scala/nl/biopet/utils/io/IoUtilsTest.scala

```diff
@@ -133,4 +145,55 @@ class IoUtilsTest extends BiopetTest {
 
     getLinesFromFile(file) shouldBe List("test")
   }
+
+  @Test
+  def testWriteStringToFile(): Unit = {
+    val string = "testing testerdetest test"
+    val file = File.createTempFile("stringtofile.", ".txt")
+    file.deleteOnExit()
+    stringToFile(string, file)
+    Source.fromFile(file).mkString shouldEqual (string + "\n")
+  }
+
+  @Test
+  def testSha256Sum(): Unit = {
+    // Taken the README from Biopet 0.9.0. Small, link should be stable
+    val downloadLink: URL =
+      new URL(
+        "https://raw.githubusercontent.com/biopet/biopet/be7838f27f3cad9f80191d92a4a795c34d1ae092/README.md")
+    getSha256SumFromDownload(downloadLink) shouldBe Some(
+      "186e801bf3cacbd564b4ec00815352218038728bd6787b71f65db474a3588901")
+    getSha256SumFromDownload(new URL(downloadLink.toString + "nonsense")) shouldBe None
+  }
+
+  @Test
+  def testListDirectory(): Unit = {
+    val tempDir = Files.createTempDirectory("test").toFile
+    tempDir.deleteOnExit()
+    val relativePaths: List[String] = List(
+      "test1.txt",
+      "test2.txt",
+      "dir1" + File.separator + "test1.txt",
+      "dir1" + File.separator + "test2.txt",
+      "dir2" + File.separator + "test1.txt",
+      "dir2" + File.separator + "test2.txt"
+    )
+    val allFiles: Seq[File] =
+      relativePaths.map(file => createTempTestFile(new File(tempDir, file)))
+
+    listDirectory(tempDir, recursive = true).toSet shouldBe allFiles.toSet
+    listDirectory(tempDir).toSet should not be allFiles.toSet
+    val twoFiles = List("test2.txt",
+                        "dir1" + File.separator + "test2.txt",
+                        "dir2" + File.separator + "test2.txt").map(file =>
+      new File(tempDir, file))
+    listDirectory(tempDir, Some(new Regex(".*2\\.txt$")), recursive = true).toSet shouldBe twoFiles.toSet
+    listDirectory(tempDir, Some(new Regex(".*2\\.txt$"))).toSet shouldBe Set(
+      new File(tempDir, "test2.txt"))
+    listDirectory(tempDir, Some(new Regex("^dir"))).toSet shouldBe Set(
+      new File(tempDir, "dir1"),
+      new File(tempDir, "dir2"))
+    listDirectory(tempDir, Some(new Regex("^dir")), recursive = true).toSet shouldBe Set()
+
```

Comment:
deleteOnExit does not work on directories. Best way doing that is making a `@AfterClass` or `@AfterSuite` method to remove it

Reply:
Done